hunts/src/filterSlice.tsx

```diff
@@ -1,53 +1,67 @@
-import { createSlice, createSelector } from "@reduxjs/toolkit";
+import { StateCreator } from "zustand";
+import { immer } from "zustand/middleware/immer"; // eslint-disable-line
+
 import { PuzzleTag } from "./types";
 import { RootState } from "./store";
 
-export const SOLVE_STATE_FILTER_OPTIONS = {
-  ALL: 1,
-  UNSOLVED: 2, // Excludes all solved puzzles and their feeders.
-  UNSOLVED_WITH_SOLVED_METAS: 3, // Includes unsolved feeders with solved metas.
-};
+export enum SOLVE_STATE_FILTER_OPTIONS {
+  ALL = 1,
+  UNSOLVED = 2, // Excludes all solved puzzles and their feeders.
+  UNSOLVED_WITH_SOLVED_METAS = 3, // Includes unsolved feeders with solved metas.
+}
+
+export interface FilterSlice {
+  filterSlice: {
+    filterValue: string;
+    solveStateFilter: SOLVE_STATE_FILTER_OPTIONS;
+    tags: PuzzleTag[];
+
+    updateFilterValue: (filterValue: string) => void;
```

Comment:
nit: For simplicity, why not call this `setFilterValue`? Update seems like it might have other side effects.

Reply:
In general, I wonder if we should use `set` instead of `update` for the various setter functions in each slice.